src/utils/set-state.ts

```diff
@@ -0,0 +1,550 @@
+
+    console.log("Writing into vault storage...");
+    await setStorageAtKey(api, storageKey.toString(), storageData);
+};
+
+const disconnectApiOnExit = <Params>(
+    api: ApiPromise,
+    callback: (params: Params) => Promise<void>
+) => async (params: Params) => {
+    await callback(params);
+    await api.disconnect();
+};
+
+// Changes deposited collateral and issued tokens of the vault to allow premium redeem
+const setPremiumRedeem = (interBtcApi: DefaultInterBtcApi) =>
+    disconnectApiOnExit(
+        interBtcApi.api,
+        async ({ accountId, collateralSymbol, wrappedSymbol }: SetVaultParamsBase) => {
+            const { api } = interBtcApi;
+            const collateralCurrency = getCurrencyFromSymbol(api, collateralSymbol) as CollateralCurrency;
+            const wrappedCurrency = getCurrencyFromSymbol(api, wrappedSymbol) as WrappedCurrency;
+            const vaultId = newVaultId(api, accountId, collateralCurrency, wrappedCurrency);
+
+            const oneBTC = "100000000";
+            const parsedIssuedTokens = api.createType("u128", oneBTC);
+            // Sets issued tokens amount to 1 BTC
+            const newIssuedTokensAmount = newMonetaryAmount<BitcoinUnit>(oneBTC, wrappedCurrency);
+            const issuedTokensModifier = (vaultData: MutableVaultData): MutableVaultData =>
+                ({ ...vaultData, issuedTokens: parsedIssuedTokens });
+
+            console.log("Setting vault issued tokens to 1 BTC...");
+            await modifyVaultData(api, vaultId, issuedTokensModifier);
+
+            // Sets vault backing collateral to average between premium redeem threshold and liquidation threshold
+            const premiumRedeemThreshold = await interBtcApi.vaults.getPremiumRedeemThreshold(collateralCurrency);
+            const liquidationThreshold = await interBtcApi.vaults.getLiquidationCollateralThreshold(collateralCurrency);
+            const newCollateralizationRatio = premiumRedeemThreshold.add(liquidationThreshold).div(2);
+            const newIssuedTokensInCollateralAmount = await interBtcApi.oracle.convertWrappedToCurrency(
+                newIssuedTokensAmount,
+                collateralCurrency as Currency<CollateralUnit>
+            );
+            const scalingFactor = new Big(Math.pow(10, FIXEDI128_SCALING_FACTOR));
+            const newCollateralAmount = newIssuedTokensInCollateralAmount.toBig().mul(newCollateralizationRatio).mul(scalingFactor);
+            const totalStakeDataHex = api.createType("u128", newCollateralAmount.toFixed()).toHex(true);
+
+            const nonceHex = api.createType("u32", 0).toHex();     // 0 nonce by default
+            const totalStakeStorageKey = getStorageMapItemKey("VaultStaking", "TotalCurrentStake", nonceHex, vaultId.toHex());
+
+            console.log("Setting backing collateral...");
+            await setStorageAtKey(api, totalStakeStorageKey, totalStakeDataHex);
+            console.log(
```

Comment:
If these setters are meant to do more than visual checking of the UI's layout given a state (I think it should), then this PR should also test that the outcomes are what you expect.

For instance, I believe these would all be tests in the UI and we must be confident they don't break because of the lib.
- `setPremiumRedeem`: test that there is at least on vault that premium redeems can be executed with, and redeem with it
- `setLiquidationVault`: test that burn redeeming works
- `setVaultReward`: test that the vault can actually claim the reward
- the other setters too

Reply:
@savudani8 do you mean to extend the commands to call the same api methods that are used in the UI or to directly test the UI components? 

Also, I have a task in this sprint to start with the UI test automation, but we will be mocking the api calls, so there won't be need to modify the parachain state (and therefore use these utils).